src/lib.rs

```diff
@@ -868,7 +871,7 @@ impl ColorScale {
         let same_position = self
             .color_stops
             .iter_mut()
-            .find(|c| position.value() == c.position.value());
+            .find(|c| (position.value() - c.position.value()).abs() < std::f64::EPSILON);
```

Comment:
no!! the `==` check on `float`s is actually desired here. Using an epsilon check would be wrong.

Reply:
Commit dropped (reverted).